### PR TITLE
feat!: default to FQCN when typename is not present

### PR DESCRIPTION
**Pull request by efgpinto**
References https://github.com/lightbend/kalix-jvm-sdk/issues/1556

This MR:
- changes the default back to FQCN when no TypeName is provided (which also decreases chances of collision within the same service)
  - this is a breaking change for Java SDK (code-first)
- update docs to advise that TypeName should ideally be unique.

### CODE REVIEW

File: sdk/java-sdk-spring/src/test/scala/kalix/javasdk/impl/JsonMessageCodecSpec.scala [hunk 1]
```diff
@@ -73,7 +73,7 @@ class JsonMessageCodecSpec extends AnyWordSpec with Matchers {
 
     "default to FQCN for typeUrl (java)" in {
       val encoded = messageCodec.encodeJava(SimpleClass("abc", 10))
-      encoded.getTypeUrl shouldBe jsonTypeUrlWith("SimpleClass")
```
Comment (octonato) [inline sdk/java-sdk-spring/src/test/scala/kalix/javasdk/impl/JsonMessageCodecSpec.scala:76]: this test was lying 😄

**Review by octonato — APPROVED** [review]
LGTM

PR description is not updated though.

File: sdk/java-sdk-spring/src/test/scala/kalix/javasdk/impl/JsonMessageCodecSpec.scala (hunk 1 above)
Comment (efgpinto) [reply sdk/java-sdk-spring/src/test/scala/kalix/javasdk/impl/JsonMessageCodecSpec.scala:76]: it was indeed :D

**Comment by efgpinto** [comment]
> PR description is not updated though.

Updated now. Also added note in docs about the recommendation to use unique type names per Kalix Service.

**Review by octonato — APPROVED** [review]

**Pull request merged**